### ISSUE TITLE
E-hentai/E-hentai Weatern Tag Issues

### ISSUE THREAD

**Issue by Conveniences67** [HDoujinDownloader/HDoujinDownloader#438]
I had setup the naming so that it would include the gallery type and I think it's having issues with the western tag, as on the site they have a sub-type of western imageset, and western comic, and if either if the two are present in the gallery it will default to unknown. Is there a way for it to default to the other two options, if they are present. Also the same might be occurring with Non-H galleries, and galleries that have been reclaassed or are in the middle of being reclassed.

**Comment by HDoujinDownloader** [comment]
You're absolutely right, it's not handling all of E-Hentai/ExHentai's gallery classifications correctly. It has to do with how HDoujin normalizes gallery types instead of using the names verbatim, and some of the labels used aren't currently recognized.

I'm making some changes on my end to address this for the next program update. Here's what I'm thinking:

"Western" galleries will be normalized to `Comic`, unless they have a different subclassification in the tags (comic/image set/etc).

"Image Set" galleries will be normalized to `Artist CG`.

"Non-H" galleries will be normalized to `Manga` (it's what the majority seem to be from what I've seen), unless they have a different subclassification in the tags (comic/image set/etc).

I'm debating whether or not I should replace the `Artist CG` label with `Image Set` instead to make it more general. Let me know if you have any thoughts on this logic/labeling.

**Comment by Conveniences67** [comment]
I can get behind the change of the western gallery tag, but when it comes to Image Set, its best to keep it as it is, not to get it confused with actual CG sets, for Non-H, I can agree with it defaulting to Manga, and Changing to a different sub class when present as this tends to occur with all the main gallery types on ex/e-hentai.

It may be best to have an option where it just has the default gallery type, then add an option in the setting to display sub type when present for example:
If Non-H  ≠ (other) then - Non-H Manga
If Non-H and (other = Art Book) then - Non-H Art Book
If Non-H and (other = Western Non-H+Comic) then  - Western Non-H Comic
If Non-H and (other = Western Non-H+Western Image Set) then  - Western Non-H Image Set

Here is a list of Sub Gallery Types for ex/e-hentai if it will help, I found them in their wiki (https://ehwiki.org/wiki/Fetish_Listing, underneath the technical and format section) :
3d
3d imageset
ai generated
anaglyph
animated
comic
figure
how to
multi-work series
non-h game manual
non-h imageset
paperchild
screenshots
sketch lines
stereoscopic
western cg
western non-h
western imageset
anthology
artbook
goudoushi
novel
soushuuhen
tankoubon
variant set
webtoon

**Comment by Conveniences67** [comment]
This update has been amazing, it has addressed most of the problems that I and others brought to your attention, please 
Keep up the good work, and continue to support and update this software.

In Regards to the %NonH% tag introduction, it would be greatly appreciated if you were to add it, as I'm currently navigating and downloading to replace a lost hard drive, and being able to quickly search in file explorer for files with that particular tag will make it easier to separate the nsfw from the sfw content.

Also, I've been trying out on the update on e-hentai/exhentai and here are the issues that I've run into:

Misc Galleries comeback as Artist CG when the option to custom name folders is enabled.

The Content of Misc. Galleries does not seem to matchup to the level of Artist CG, especially when some of the Misc contains, images of paper cut outs, dolls, or game manuals as shown here:
https://ehwiki.org/wiki/Gallery_Categories

A Differential for Misc would be appreciated, as their are a lot of galleries which either contain ai generated content, or content with 3d models, or its just a themeless imageset masquerading as a misc gallery. It would be greatly appreciated if the tags AI, 3D, and Misc be utilized to help users quickly glance at the folder/file's name and decided whether or not they want to view the content 

I also tested out whether or not a bookmark link from an older gallery, will self-update to a new one, and your right it's currently not working.

Is their also a way to update a previously downloaded gallery, and replace it with the contents of the updated one, I've been looking into the source code for the site of older galleries to help out, and one way I found, is to search for "newer versions of gallery available" and to open and download from the newest link, in case their are multiple updates, 
maybe have it compared to the download date of the previous one, to help the program determine when an update is needed

As Currently I have the file naming setup to were the timestamp is displayed (*Which has its own problems) to help me identify when a gallery has been updated or a better version has ben found and needs to be replaced.

*The default format, is too long for a quick glance and comparison between two similar filenames, is it possible to replace the timestamp with something like this: %date:format=yyyy-MM-dd%, so people can customize and choose the way they view their preferred date formats

d	the day as number without a leading zero (1 to 31)
dd	the day as number with a leading zero (01 to 31)
ddd	the abbreviated localized day name
dddd	the long localized day name
M	the month as number without a leading zero (1-12)
MM	the month as number with a leading zero (01-12)
MMM	the abbreviated localized month name
MMMM	the long localized month name
yy	the year as two digit number (00-99)
yyyy	the year as four digit number
h	the hour without a leading zero (0 to 23 or 1 to 12 if AM/PM display)
hh	the hour with a leading zero (00 to 23 or 01 to 12 if AM/PM display)
m	the minute without a leading zero (0 to 59)
mm	the minute with a leading zero (00 to 59)
s	the second without a leading zero (0 to 59)
ss	the second with a leading zero (00 to 59)
z	the milliseconds without leading zeroes (0 to 999)
zzz	the milliseconds with leading zeroes (000 to 999)
AP	use AM/PM display. AP will be replaced by either "AM" or "PM".
ap	use am/pm display. ap will be replaced by either "am" or "pm".
